datasets/dataset.py

```diff
@@ -178,3 +178,20 @@ def get_input(self, image_name, data_dir, image_data_fmt, input_shape):
 		Return the input corresponding to one image file
 		"""
 		return get_resized_image(image_name, data_dir, image_data_fmt, input_shape) / 255.0
+
+
+	def generate_tta(self, inputs):
+		"""
+		Generate augmented data sets for more stable prediction
+		"""
+		tta = []
+		image_data = inputs if self.input_length == 1 else inputs[0]
+		for i in range(4):
+			tta.append(rotate_images(inputs, 90 * i))
+		if self.input_length == 1:
```

Comment:
The input length thing is a bit confusing. You can ignore it and consider that input_length == 1
(for datasets with inputs like [image, extradata], the image data is the first element of the input, else it's the whole input)